packages/core/src/NavigationContainer.tsx

```diff
@@ -88,9 +88,7 @@ const Container = React.forwardRef(function NavigationContainer(
 
   const dispatch = (action: NavigationAction) =>
     context.performTransaction(() => {
-      for (let i = 0; i < actionListeners.length; i++) {
```

Comment:
why this change?

Reply:
oh I get it, cause there can only be one navigator, right?

Reply:
I'd keep the loop. the container shouldn't know that each navigator always adds only one listener

Reply:
why? I see it needed bc we will need to return this value soon while wanted to implement back behavior